monix-execution/shared/src/main/scala/monix/execution/FutureUtils.scala

```diff
@@ -161,6 +172,10 @@ object FutureUtils {
     p.future
   }
 
+  /** Creates a future that never completes. */
+  def never[A]: Future[A] =
+    Promise[A].future
```

Comment:
The other option would be copy-pasting this from Scala into Monix, but I'm not sure whether that's a good idea.